app/src/androidTest/java/org/mozilla/focus/activity/URLCompletionTest.java

```diff
@@ -67,4 +92,208 @@ public void CompletionTest() throws InterruptedException, UiObjectNotFoundExcept
         TestHelper.hint.waitForExists(waitingTime);
         assertTrue (TestHelper.inlineAutocompleteEditText.getText().equals("http://www.mozilla.org"));
     }
+
+    @Test
+    public void CustomCompletionTest() throws UiObjectNotFoundException {
+        OpenCustomCompleteDialog();
+
+        // Enable URL autocomplete, and add URL
+        toggleAutocomplete();
+        addAutoComplete(site);
+        exitToTop();
+
+        // Check for custom autocompletion
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.setText("68");
+        TestHelper.hint.waitForExists(waitingTime);
+        assertTrue(TestHelper.inlineAutocompleteEditText.getText().equals(site));
+        TestHelper.cleartextField.click();
+
+        // Remove custom autocompletion site
+        OpenCustomCompleteDialog();
+        removeACSite();
+        exitToTop();
+
+        // Check autocompletion
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.setText("68");
+        TestHelper.hint.waitForExists(waitingTime);
+        assertTrue(TestHelper.inlineAutocompleteEditText.getText().equals("68"));
+        TestHelper.cleartextField.click();
+
+        // Reset to disabled
+        OpenCustomCompleteDialog();
+        toggleAutocomplete();       // Disable autocomplete
+    }
+
+    @Test
+    // add custom autocompletion site, but disable autocomplete
+    public void DisableCCwithSiteTest() throws UiObjectNotFoundException {
```

Comment:
\[PMD\] INFO: The method 'DisableCCwithSiteTest\(\)' has a NCSS line count of 26\.